test/kitchensinke2e/broker_test.go

```diff
@@ -0,0 +1,132 @@
+	"knative.dev/reconciler-test/pkg/feature"
+)
+
+func BrokerReadiness(testLabel string, broker component, brokerDls component, triggers []component, triggerDls component) *feature.Feature {
+	brokerName := testLabel
+	dlsName := testLabel + "-dls"
+	triggerDlsName := testLabel + "-tdls"
+	f := feature.NewFeatureNamed(fmt.Sprintf("%s with %s Broker dls and %s Trigger dls", label(broker), label(brokerDls), label(triggerDls)))
+
+	if triggerDls != nil {
+		f.Setup("Install A Trigger DLS Thing", triggerDls.Install(triggerDlsName))
+	}
+
+	if brokerDls != nil {
+		f.Setup("Install A Broker DLS Thing", brokerDls.Install(dlsName))
+		f.Setup("Install A Broker with Delivery", broker.Install(brokerName, delivery.WithDeadLetterSink(brokerDls.KReference(dlsName), "")))
+	} else {
+		f.Setup("Install A Broker", broker.Install(brokerName))
+	}
+
+	for _, trigger := range triggers {
+		// We'll label the Trigger based on the trigger component label, which should be unique
+		triggerName := brokerName + "-" + trigger.ShortLabel()
+
+		f.Setup(fmt.Sprintf("Install a %s Trigger Thing", trigger.Label()), trigger.Install(triggerName))
+
+		if triggerDls != nil {
+			f.Setup(fmt.Sprintf("Install a %s Trigger", trigger.Label()), triggerresources.Install(
+				triggerName,
+				brokerName,
+				triggerresources.WithSubscriber(trigger.KReference(triggerName), ""),
+				triggerresources.WithDeadLetterSink(triggerDls.KReference(triggerDlsName), "")))
+		} else {
+			f.Setup(fmt.Sprintf("Install a %s Trigger", trigger.Label()), triggerresources.Install(
+				triggerName,
+				brokerName,
+				triggerresources.WithSubscriber(trigger.KReference(triggerName), "")))
+		}
+	}
+
+	f.Assert("Broker is Ready", broker.IsReady(brokerName))
+	for _, trigger := range triggers {
+		triggerName := brokerName + "-" + trigger.ShortLabel()
+		f.Assert(fmt.Sprintf("Trigger %s is Ready", trigger.Label()), triggerresources.IsReady(triggerName))
+	}
+	return f
+}
+
+func TestBrokerReadiness(t *testing.T) {
+	brokers := []component{
+		inMemoryChannelMtBroker,
+		kafkaChannelMtBroker,
+		kafkaBroker,
+	}
+
+	deadLetterSinks := []component{
+		kafkaChannel,
+		inMemoryChannel,
+		genericChannelWithKafkaChannelTemplate,
+		genericChannelWithInMemoryChannelTemplate,
+		inMemoryChannelMtBroker,
+		kafkaChannelMtBroker,
+		kafkaBroker,
+		inMemoryChannelSequence,
+		kafkaChannelSequence,
+		inMemoryChannelParallel,
+		kafkaChannelParallel,
+		ksvc,
+	}
+
+	triggers := []component{
+		kafkaChannel,
+		inMemoryChannel,
+		genericChannelWithKafkaChannelTemplate,
+		genericChannelWithInMemoryChannelTemplate,
+		inMemoryChannelMtBroker,
+		kafkaChannelMtBroker,
+		kafkaBroker,
+		inMemoryChannelSequence,
+		kafkaChannelSequence,
+		inMemoryChannelParallel,
+		kafkaChannelParallel,
+		ksvc,
+	}
+
+	// Test all combinations of Broker X DeadLetterSinks, each broker with all possible Triggers
+	// with the DeadLetterSink set on the Broker
+	for _, broker := range brokers {
```

Comment:
I don't know why but this file (and a few others) shows different indentation that for example test/kitchensinke2e/channel_test.go